libs/src/services/discoveryProvider/index.js

```diff
@@ -82,9 +82,16 @@ class DiscoveryProvider {
     this.discoveryProviderEndpoint = endpoint
   }
 
-  async autoSelectEndpoint (retries = 3) {
+  /**
+   * Wrapper method to auto select a valid discovery provider.
+   * @param {*} retries max retries before throwing an error
+   * @param {*} clearDPLocalStorageEntryAndClearInterval if set to true, implies that the previously
+   * selected discovery provider has been failing to serve requests. The prior recurring interval of
+   * checking local storage for DP and old DP local storage entry need to be cleared.
+   */
+  async autoSelectEndpoint (retries = 3, clearDPLocalStorageEntryAndClearInterval = false) {
```

Comment:
since we don't use the abbreviation DP anywhere in the app--how about

clearCachedDiscoveryProvider = false

clearing the interval seems implicit with clearing the thing in cache anyway